pkg/apis/config/feature_flags.go

```diff
@@ -138,6 +142,24 @@ func GetFeatureFlagsConfigName() string {
 	return "feature-flags"
 }
 
+func getEnforceNonfalsifiabilityFeature(cfgMap map[string]string, feature *string) error {
+	var mapValue struct{}
+	var acceptedValues = map[string]struct{}{
+		EnforceNonfalsifiabilityNone:      mapValue,
+		EnforceNonfalsifiabilityWithSpire: mapValue,
+	}
+	var value = DefaultEnforceNonfalsifiability
+	if cfg, ok := cfgMap[enforceNonfalsifiability]; ok {
+		value = strings.ToLower(cfg)
+	}
+	if _, ok := acceptedValues[value]; !ok {
+		*feature = DefaultEnforceNonfalsifiability
+		return fmt.Errorf("invalid value for feature flag %q: %q", enforceNonfalsifiability, value)
+	}
+	*feature = value
```

Comment:
the interface here might be a bit simpler if you returned the string value to use vs. changing the value of a pointer (i.e. more of a functional interface vs. a function with side effects)

i think this might also simplify the code below where you check if SPIRE is configured in non-alpha mode; all you need to do is assert that the string returned is 'none'/ no error

Reply:
+1,  changed to a functional interface.